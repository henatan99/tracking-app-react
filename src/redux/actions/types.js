export const LOGIN_USER = 'LOGIN_USER';
export const LOGOUT_USER = 'LOGOUT_USER';
export const SET_MEASUREMENTS = 'SET_MEASUREMENTS';
export const CREATE_MEASUREMENT = 'CREATE_MEASUREMENT';
