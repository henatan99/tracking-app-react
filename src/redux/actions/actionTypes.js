export const CHANGE_USERNAME = 'CHANGE_USERNAME'