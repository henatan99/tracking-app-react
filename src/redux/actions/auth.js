import {
    REGISTER_SUCCESS,
    REGISTER_FAIL,
    LOGIN_SUCCESS,
    LOGIN_FAIL,
    LOGOUT,
    SET_MESSAGE,
} from './actionTypes';

import AuthService from '../services/auth.service';
import username from '../reducers/username';

export const register = (username) => (dispatch) => {
    return AuthService.register(username)
        .then((response) => {
                dispatch({
                    type: REGISTER_SUCCESS,
                });
                
                dispatch({
                    type: SET_MESSAGE,
                    payload: response.data.message,
                });

                return Promise.resolve();
            },

            (error) => {
                const message = (
                    error.response &&
                    error.response.data &&
                    error.response.data.message) ||
                    error.message || error.toString();

                dispatch({
                    type: REGISTER_FAIL,
                });

                dispatch({
                    type: SET_MESSAGE,
                    payload: message,
                });

                return Promise.reject();                
            }
        ) 
}

export const login = (username) => (dispatch) => {
    return AuthService.login(username).then(
        (data) => {
            dispatch({
                type: LOGIN_SUCCESS,
                payload: { user: data },
            });

            return Promise.resolve();
        },

        (error) => {
            const message = 
            (error.response &&
                error.response.data &&
                error.response.data.message) ||
            error.message ||
            error.toString();

            dispatch({
                type: LOGIN_FAIL,
            });

            dispatch({
                type: SET_MESSAGE,
                payload: message,
            })

            return Promise.reject();
        }
    )
};

export const logout = () => (dispatch) => {
    AuthService.logout();

    dispatch({
        type: LOGOUT,
    });
};