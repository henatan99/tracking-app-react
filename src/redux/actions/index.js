import axios from 'axios';

import {
  SIGNUP_USER,
  LOGIN_USER,
  LOGOUT_USER,
  SET_MEASUREMENTS,
  CREATE_MEASUREMENT,
  CREATE_GOAL,
  FETCH_MEASUREDS_REQUEST,
  FETCH_MEASUREDS_SUCCESS,
  FETCH_MEASUREDS_FAILURE,
  FETCH_FILTERED_MEASUREDS_REQUEST,
  FETCH_FILTERED_MEASUREDS_SUCCESS,
  FETCH_FILTERED_MEASUREDS_FAILURE,
} from './types';

export const signupUser = (user) => ({
  type: SIGNUP_USER,
  payload: user,
});

export const loginUser = (user) => ({
  type: LOGIN_USER,
  payload: user,
});

export const logoutUser = () => ({
  type: LOGOUT_USER,
});

export const setMeasurements = (measurements) => ({
  type: SET_MEASUREMENTS,
  payload: measurements,
});

export const createMeasurement = (measured) => ({
  type: CREATE_MEASUREMENT,
  payload: measured,
});

export const createGoal = (goal) => ({
  type: CREATE_GOAL,
  payload: goal,
});

export const fetchMeasureds = (userId) => async (dispatch) => {
  dispatch({ type: FETCH_MEASUREDS_REQUEST });

  try {
    const token = localStorage.getItem('token');
    const response = await axios.get(
      `https://pure-tundra-23506.herokuapp.com/users/${userId}/measureds`,
      {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      },
    );

    dispatch({ type: FETCH_MEASUREDS_SUCCESS, payload: response.data });
  } catch (error) {
    dispatch({ type: FETCH_MEASUREDS_FAILURE }, error);
  }
};

export const fetchfilterByMeasurementIdMeasureds = (userId) => async (dispatch) => {
  dispatch({ type: FETCH_FILTERED_MEASUREDS_REQUEST });

  try {
    const token = localStorage.getItem('token');
    const response = await axios.get(
      `https://pure-tundra-23506.herokuapp.com/users/${userId}/filter_by_measurement_id_measureds`,
      {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      },
    );

    console.log(response);

    dispatch({ type: FETCH_FILTERED_MEASUREDS_SUCCESS, payload: response.data });
  } catch (error) {
    dispatch({ type: FETCH_FILTERED_MEASUREDS_FAILURE }, error);
  }
};
