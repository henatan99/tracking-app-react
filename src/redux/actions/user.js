import { LOGIN_USER, LOGOUT_USER } from './types';

export const loginUser = (user) => ({
  type: LOGIN_USER,
  payload: user,
});

export const logoutUser = (user) => ({
  type: LOGOUT_USER,
  payload: user,
});
