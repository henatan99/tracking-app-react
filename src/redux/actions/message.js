import { SET_MESSAGE, CLEAR_MESSAGE } from './actionTypes'

export const setMessage = (message) => ({
    type: SET_MESSAGE,
    payload: message,
});

export const clearMessage = () => ({
    type: CLEAR_MESSAGE,
});