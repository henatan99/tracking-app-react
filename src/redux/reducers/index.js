import { combineReducers } from 'redux';
import user from './measurements';
import measurements from './user';
import measureds from './measureds';
import filteredMeasureds from './filteredMeasureds';

export default combineReducers({
  user, measurements, measureds, filteredMeasureds,
});
