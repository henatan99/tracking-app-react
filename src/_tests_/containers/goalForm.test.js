import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import GoalForm from '../../containers/goalForm';
import { loginUser } from '../../redux/actions';

const mockStore = configureStore([]);

describe('LoginForm', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      myState: 'sample text',
    });

    store.dispatch = jest.fn();

    component = renderer.create(
      <Provider store={store}>
        <GoalForm />
      </Provider>,
    );
  });

  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should dispatch an action on button click', () => {
    renderer.act(() => {
      component.root.findByType('form').props.onSubmit();
    });

    expect(store.dispatch).toHaveBeenCalledTimes(1);
    expect(store.dispatch).toHaveBeenCalledWith(
      loginUser({ payload: 'sample text' }),
    );
  });
});
