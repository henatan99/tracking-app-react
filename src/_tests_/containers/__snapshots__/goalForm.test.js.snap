// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should render with given state from Redux store 1`] = `
<div
  className="login"
>
  <h1
    className="login-header"
  >
    Login Page
  </h1>
  <form
    className="login-form"
    onSubmit={[Function]}
  >
    <input
      className="login-input"
      onChange={[Function]}
      placeholder="username"
      type="text"
      value=""
    />
    <br />
    <button
      className="login-button"
      type="submit"
    >
      Login
    </button>
    <span>
      
    </span>
  </form>
</div>
`;
